ui/pages/confirmations/components/confirm/info/approve/hooks/use-is-nft.ts

```diff
@@ -0,0 +1,23 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useEffect, useState } from 'react';
+import { getTokenStandardAndDetails } from '../../../../../../../store/actions';
+
+export const useIsNFT = (
+  transactionMeta: TransactionMeta,
+): { isNFT: boolean } => {
+  const [decimals, setDecimals] = useState('');
+  useEffect(() => {
+    const fetchTokenDetails = async () => {
+      const result = await getTokenStandardAndDetails(
```

Comment:
As we rely on this in a lot of places, would it have even greater value to wrap the entire thing in a hook such as `useTokenStandardAndDetails`?

Reply:
The hook is so much more simplified with `useAsyncResult` that I think I'll hold off creating one for `getTokenStandardAndDetails`. That can be done later in a separate PR to avoid increasing the scope of this one.